src/pymor/algorithms/dmd.py

```diff
@@ -0,0 +1,270 @@
+        (`'method_of_snapshots'` or `'qr_svd'`).
+
+
+    Returns
+    -------
+    Wk : VectorArray
+        |VectorArray| containing the dynamic modes.
+
+    omega : array_like
+        Time scaled eigenvalues: `ln(l)/dt`.
+    """
+
+    assert modes in ('exact', 'standard', 'exact_scaled')
+    assert A is None or (XL is None and XR is None)
+    assert not(A is None and (XL and XR is None))
+    assert isinstance(A, VectorArray) or A is None
+    assert isinstance(XL, VectorArray) or XL is None
+    assert isinstance(XR, VectorArray) or XR is None
+    assert A is None or (target_rank is None or target_rank <= len(A))
+    assert XL is None or (target_rank is None or target_rank < len(XL))
+    assert XL is None and XR is None or len(XL) == len(XR)
+    assert order in (True, False)
+    assert svd_method in ('qr_svd', 'method_of_snapshots')
+
+    # XL = x0, ..., xm-1  ; XR = x1, ..., xm
+    if A is not None:
+        XL = A[:-1]
+        XR = A[1:]
+
+    rank = len(XL) if target_rank is None else target_rank
+    svd = qr_svd if svd_method == 'qr_svd' else method_of_snapshots
+
+    print('SVD of XL...')
+    U, SVALS, Vh = svd(XL, product=None, modes=rank)
+
+    # Invert the Singular Values
+    SVALS_diag = np.diag(SVALS)
+    Sigma_inv = np.linalg.inv(SVALS_diag)
+    V = Vh.conj().T
+
+    # Solve the least Sqaures Problem
+    # real: A_tilde = U.T * XR * Vh.T * Sigma_inv
+    # complex: A_tilde = U.H * XR * Vh.H * Sigma_inv
+    # A = XR[:k] @ V @ Sigma_inv.inner(Uh)
+    A_tilde = U.inner(XR) @ V @ Sigma_inv
+
+    print('Calculating eigenvalue dec. ...')
+    EVALS, EVECS = spla.eig(A_tilde, b=None, left=False, right=True)
+    # omega = np.log(EVALS) / dt
+
+    omega = EVALS / dt
+
+    if order:
+        print('Sorting...')
+        # return ordered result
+        sort_idx = np.argsort(np.abs(omega))[::-1]
+        EVECS = EVECS[:, sort_idx]
+        EVALS = EVALS[sort_idx]
+        omega = omega[sort_idx]
+
+    print('Reconstructing Eigenvectors...')
+    if modes == 'standard':
+        Wk = U.lincomb(EVECS.T)
+    elif modes == 'exact' or 'exact_scaled':
+        Wk = XR.lincomb(V.T).lincomb(Sigma_inv.T).lincomb(EVECS.T)
+        if modes == 'exact_scaled':
+            EVALS_inv = np.reciprocal(EVALS)
+            Wk = Wk*EVALS_inv
+
+    return Wk, omega
+
+
+def rand_QB(A, target_rank=None, distribution='normal', oversampling=0, powerIterations=0):
+    """
+    randomisierte QB-Zerlegung
+
+    See Algorithm 3.1 in [EMKB19]_.
+
+    Parameters
+    ----------
+    A  :
+        The |VectorArray| for which the randomized QB Decomposition is to be computed.
+
+    target_rank  :  int
+        The desired rank for the decomposition. If None rank = len(A).
+
+    distribution : str
+        Distribution used for the random projectionmatrix Omega. (`'normal'` or `'uniform'`)
+
+    oversampling : int
+        Oversamplingparameter. Number of extra columns of the projectionmatrix.
```

Comment:
```suggestion
        Oversampling parameter. Number of extra columns of the sample matrix.
```

Is the term 'projection matrix' used in the literature? If any of the matrices, I would call 'Q' the projection matrix, not omega.